core/drivers/crypto/caam/cipher/caam_cipher_xts.c

```diff
@@ -0,0 +1,254 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2018-2020 NXP
+ *
+ * Implementation of Cipher XTS functions
+ */
+#include <caam_common.h>
+#include <caam_utils_mem.h>
+#include <mm/core_memprot.h>
+#include <string.h>
+
+#include "local.h"
+
+/*
+ * Galois Multiplication
+ *
+ * @buf  [in/out] buffer to multiply
+ */
+static void do_galois_mult(struct caambuf *buf)
+{
+	size_t idx = 0;
+	uint8_t tmp = 0;
+	uint8_t tmptmp = 0;
+
+	for (idx = 0; idx < buf->length; idx++) {
+		tmptmp = buf->data[idx] >> 7;
+		buf->data[idx] = ((buf->data[idx] << 1) | tmp) & UINT8_MAX;
```

Comment:
minor: mask not needed as `buf->data[idx]` and `tmp` are all `uint8_t`.

Reply:
I will drop the mask